students/tri_nguyen/lesson10/mailroom_fn.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------- #
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  18-May-2018
+# ------------------------------------------- #
+
+# Import stuff
+from operator import itemgetter
+import time
+from pprint import pprint
+
+# Get local time
+local_time = time.localtime()
+
+# Declare variables and data structure
+
+# donors dict stores each donation from each donor
+donors = {
+
+    'William Gates, III': [326892.24, 396892.24, 600],
+    'Mark Zuckerberg': [546.37, 5485.37, 6465.37, 450],
+    'Jeff Bezos': [877.33, 898],
+    'Paul Allen': [235.14, 216.14, 296.14],
+    'Tri Nguyen': [100]
+    }
+
+# Declare functions
+
+def total(donations):
+    ''' return the total amount of money a donor has donated '''
+    return sum(donations)
+
+
+def num_gifts(donations):
+    ''' return how many times a donor has donated '''
+    return len(donations)
+
+
+def average(donations):
+    ''' return the average of all the amount 
+        of money that a donor has donated '''
+    return sum(donations) / len(donations)
```

Comment:
when you have several functions with the same input param it is usually indication that this should be a class